# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/WakeLockListener.cpp
# Commit: 5db268e2a934
# Full Hash: 5db268e2a9349aad62c83a819169aeeca62d9973
# Author: Greg V <greg@unrelenting.technology>
# Date: 2020-01-09 09:44:15
# Description:
#   Bug 1607368 - check for headless when checking for Wayland idle-inhibit r=stransky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59094
# ==============================================================================

diff -r adeaa48edc50 -r 5db268e2a934 widget/gtk/WakeLockListener.cpp
--- a/widget/gtk/WakeLockListener.cpp	Thu Jan 09 00:20:47 2020 +0000
+++ b/widget/gtk/WakeLockListener.cpp	Thu Jan 09 00:12:37 2020 +0000
@@ -222,7 +222,12 @@
 
 /* static */
 bool WakeLockTopic::CheckWaylandIdleInhibitSupport() {
+  // Gtk is not guaranteed to be built with Wayland support, so we have to
+  // check for every non-Wayland backend instead of checking for Wayland
+  if (gfxPlatform::IsHeadless()) return false;
+
   GdkDisplay* gDisplay = gdk_display_get_default();
+  if (!gDisplay) return false;
   if (GDK_IS_X11_DISPLAY(gDisplay)) return false;
 
   widget::nsWaylandDisplay* waylandDisplay =
